### ISSUE TITLE
ORCID search result list ranking

### ISSUE THREAD

**Issue by sergejzr** [gdcc/dataverse-external-vocab-support#8]
Dear Dataverse Vocab Community!

It appears that the ORCID result list should be reversed when the user enters a query (maybe also other lists in general)

Example: search for my name "Sergej Zerr" (also any other names I tested)

![image](https://user-images.githubusercontent.com/12542792/140528360-19052983-a7da-41e0-8609-256c95ff4fbb.png)

**Comment by qqmyers** [comment]
Thanks for investigating! 

I think I've found the issue. FWIW, it is not as simple as it seems - ORCID controls the ordering of the initial results and they rank on more than name and then, once you've selected a name, your selections are cached in browser storage, and it is supposed to appear at the top of the list whenever ORCID returns it (so on subsequent runs "Sergej" should be enough to pull yours to the top.)

In any case, it looks like the sort to raise the rank of previously used ORCIDs was also flipping the order in the rest of the list. The latest commit should fix that. Again - I haven't tested but if you're using the github.io javascripts you should now see the results in the right order.

**Comment by sergejzr** [comment]
Thanks for considering, @qqmyers,
still the behavior is a bit strange. If I start typing the name, until "Sergej Zer" the correct name even does not show up at all. When I complete the last letter, it suddenly appears (and this time correctly at a top)

**Comment by qqmyers** [comment]
Interesting. FWIW: The call being made is URL: https://pub.sandbox.orcid.org/v3.0/expanded-search/?q=`<your string>` (requesting application/json and limiting the results to 10, but you should be able to use the URL above in your browser to see what ORCID returns. ) If they are doing something like only matching whole words, there's probably not much we can do but they might take the feedback that partial matches would be valuable.

Given that the order is now correct (thanks again for checking!), I'll close this issue. If you find anything we can do on our side/in Javascript to improve the results, feel free to open an issue and/or submit a PR. (The current scripts were intended as initial examples, not static final products.)

**Comment by sergejzr** [comment]
I see, thank you! I did some testing of the API and, definitely, the issue is on ORCIDs site, the search there needs some improvement. 

There a few things I found out:
(1) They are writing that the [sandbox just contains test data](https://info.orcid.org/documentation/integration-guide/sandbox-testing-server/) and a full fledged API would be available for member accounts. Curious if the search is better there. 
(2) There is also a [public data file ](https://info.orcid.org/documentation/integration-guide/working-with-bulk-data/)and search for newly changed records - meaning one could maintain the own registry if needed. 
(3) The search API allows for pagination and Lucene query language (public API is limited to 10 000 search results)

Unfortunately the API seems not to work reliably at the moment (nothing is returned for my name, though it worked an hour ago). I will try in the next days. In case Lucene queries are supported properly, we can have a hope :) For productive environments, Dataverse should probably allow to use membership accounts for the API.
